module/VuFind/src/VuFind/View/Helper/Root/CleanHtmlFactory.php

```diff
@@ -0,0 +1,156 @@
+     * @throws ServiceNotCreatedException if an exception is raised when
+     * creating a service.
+     * @throws ContainerException if any other error occurs
+     */
+    public function __invoke(
+        ContainerInterface $container,
+        $requestedName,
+        ?array $options = null
+    ) {
+        if (!empty($options)) {
+            throw new \Exception('Unexpected options sent to factory.');
+        }
+
+        $this->container = $container;
+        return new $requestedName(Closure::fromCallable([$this, 'createPurifier']));
+    }
+
+    /**
+     * Create a purifier instance.
+     *
+     * N.B. This is a relatively slow method.
+     *
+     * @param array $options Additional options. Currently supported:
+     *   bool   targetBlank Whether to add target="_blank" to external links
+     *   string context     Rendering context ('default' or 'heading')
+     *
+     * @return HTMLPurifier
+     */
+    protected function createPurifier(array $options): HTMLPurifier
+    {
+        $config = \HTMLPurifier_Config::createDefault();
+        // Set cache path to the object cache
+        $cacheDir
+            = $this->container->get(\VuFind\Cache\Manager::class)->getCache('object')->getOptions()->getCacheDir();
+        if ($cacheDir) {
+            $config->set('Cache.SerializerPath', $cacheDir);
+        }
+        if ($options['targetBlank'] ?? false) {
+            $config->set('HTML.Nofollow', 1);
+            $config->set('HTML.TargetBlank', 1);
+        }
+
+        // Setting the following option makes purifier’s DOMLex pass the
+        // LIBXML_PARSEHUGE option to DOMDocument::loadHtml method. This in turn
+        // ensures that PHP calls htmlCtxtUseOptions (see
+        // github.com/php/php-src/blob/PHP-8.1.14/ext/dom/document.c#L1870),
+        // which ensures that the libxml2 options (namely keepBlanks) are set up
+        // properly, and whitespace nodes are preserved. This should not be an
+        // issue from libxml2 version 2.9.5, but during testing the issue was
+        // still intermittently present, and this setting should remain in place.
+        $config->set('Core.AllowParseManyTags', true);
+
+        $this->setAdditionalConfiguration($config, $options);
+        return new \HTMLPurifier($config);
+    }
+
+    /**
+     * Sets additional configuration
+     *
+     * @param HTMLPurifier_Config $config  Configuration
+     * @param array               $options Additional options
+     *
+     * @return void
+     */
+    protected function setAdditionalConfiguration(HTMLPurifier_Config $config, array $options): void
+    {
+        $context = $options['context'] ?? 'default';
+        if ('heading' === $context) {
+            // Limit elements allowed in headings
+            // (see https://developer.mozilla.org/en-US/docs/Web/HTML/Content_categories#phrasing_content for more
+            // information):
+            $config->set('HTML.AllowedElements', 'a,b,br,em,i,span');
```

Comment:
Should we make this configurable? If not, should we at least reference this factory in config.ini so people know what "sanitized HTML" actually means?